tachyon/zk/transcript/poseidon_transcript.h

```diff
@@ -0,0 +1,129 @@
+// Copyright 2020-2022 The Electric Coin Company
+// Copyright 2022 The Halo2 developers
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.halo2 and the LICENCE-APACHE.halo2
+// file.
+
+#ifndef TACHYON_ZK_TRANSCRIPT_POSEIDON_TRANSCRIPT_H_
+#define TACHYON_ZK_TRANSCRIPT_POSEIDON_TRANSCRIPT_H_
+
+#include <utility>
+#include <vector>
+
+#include "tachyon/base/buffer/buffer.h"
+#include "tachyon/base/buffer/vector_buffer.h"
+#include "tachyon/crypto/hashes/sponge/poseidon/halo2_poseidon.h"
+#include "tachyon/zk/transcript/transcript.h"
+
+namespace tachyon::zk {
+
+template <typename AffinePointTy>
+class PoseidonRead : public TranscriptRead<AffinePointTy> {
+ public:
+  using ScalarField = typename AffinePointTy::ScalarField;
+  using Buffer = typename base::Buffer;
```

Comment:
```suggestion
  using Buffer = base::Buffer;
```

Plus, if it is this much short, i think it's better to use a plain `base::Buffer`.